tests/infra/ccf.py

```diff
@@ -90,6 +97,12 @@ def __init__(
         self.nodes = []
         self.hosts = hosts
         self.status = ServiceStatus.CLOSED
+        self.binary_dir = binary_dir
+        self.key_generator = os.path.join(binary_dir, self.KEY_GEN)
```

Comment:
Since we use 3.7, if we wanted to be fancy, we could use pathlib (https://docs.python.org/3/library/pathlib.html), tell argparse that `args.binary_dir` is a `Path`, and then do stuff like `self.key_generator = binary_dir / self.KEY_GEN`.

I don't know that it'd be worth it, but we could.

Reply:
On the one hand, I'm all for some semblance of type protections. But on the other, I hate this overload of the `/` operator. My vote is a mild no. I think if we're going to use it at all, it should be a separate PR to replace all our `os.path` useage.